cli_tools_tests/e2e/gce_windows_upgrade/test_suites/windows_upgrade/windows_upgrade_test_suite.go

```diff
@@ -87,6 +88,20 @@ func TestSuite(
 		testsMap[testType][testBYOL] = runWindowsUpgradeBYOL
 	}
 
+	// These tests only apply to wrapper since it's testing staging install media.
+	staging2012 := junitxml.NewTestCase(
+		testSuiteName, fmt.Sprintf("[%v] Staging test 2008 -> 2012", e2e.Wrapper))
+	staging2016 := junitxml.NewTestCase(
+		testSuiteName, fmt.Sprintf("[%v] Staging test 2012 -> 2016", e2e.Wrapper))
+	staging2019 := junitxml.NewTestCase(
```

Comment:
What about the following upgrade paths:
- 2016 to 2019
- 2016 to 2022
- 2019 to 2022

Reply:
the purpose is to validate that the image covers install media for 2016/2019/2022. It doesn't matter what is the old version. I chose 2012 for all tests to reduce the base test image.

Reply:
Changed to cover 2012->2016,2016->2019,2019->2022. It can guarantee that all versions' media is complete. No need to cover all combinations in my opinion.